BoxContentSDK/BoxContentSDK/BOXNSURLSessionManager.m

```diff
@@ -250,6 +280,23 @@ - (void)URLSession:(NSURLSession *)session task:(NSURLSessionTask *)task
     [self dessociateSessionTaskId:task.taskIdentifier];
 }
 
+/* Sent if a task requires a new, unopened body stream.  This may be
+ * necessary when authentication has failed for any request that
+ * involves a body stream.
+ */
+- (void)URLSession:(NSURLSession *)session task:(NSURLSessionTask *)task
+ needNewBodyStream:(void (^)(NSInputStream * _Nullable bodyStream))completionHandler
+{
+    NSInputStream *inputStream = nil;
+
+    if (task.originalRequest.HTTPBodyStream && [task.originalRequest.HTTPBodyStream conformsToProtocol:@protocol(NSCopying)]) {
+        inputStream = [task.originalRequest.HTTPBodyStream copy];
+    }
+
+    if (completionHandler) {
```

Comment:
So we expect to always have a copy-able stream... Should we at least log something like an error or warning if that is not the case? Like if we're in this if and the inputStream is nil. Otherwise corner cases might be harder to debug.